### PR TITLE
Adds command for updating builder.toml

### PR DESCRIPTION
**Pull request by ForestEckhardt**
## Summary

A `jam` command that will update a user `builder.toml` to use the latest versions of the buildpacks, lifecycle, and build image.

## Use Cases

Maintaining a `builder.toml`.

Partially solves the following [issue](https://github.com/paketo-buildpacks/github-config/issues/187).

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have added an integration test, if necessary.

### CODE REVIEW

File: cargo/jam/internal/image.go [hunk 1]
```diff
@@ -56,3 +57,59 @@ func FindLatestImage(uri string) (Image, error) {
 		Version: versions[len(versions)-1].String(),
 	}, nil
 }
+
+func FindLatestBuildImage(runURI, buildURI string) (Image, error) {
```
Comment (sophiewigmore) [inline cargo/jam/internal/image.go:61]: Making sure I understand this - this function is to get the build image that shares the same tag as the run image?

Comment (ForestEckhardt) [reply cargo/jam/internal/image.go:61]: Yes!

**Comment by sophiewigmore** [comment]
Code looks good and mimics what we do in the builder update GHA really well. Tested this out locally and works as expected

**Comment by ForestEckhardt** [comment]
@ryanmoran When you have a chance to take a look!

File: go.mod [hunk 2]
```diff
@@ -5,6 +5,7 @@ go 1.13
 require (
 	github.com/BurntSushi/toml v0.3.1
 	github.com/Masterminds/semver v1.5.0
+	github.com/Masterminds/semver/v3 v3.1.1
```
Comment (ryanmoran) [inline go.mod:8]: Do we need 2 different versions of this library?

Comment (ForestEckhardt) [reply go.mod:8]: The v3 version branch allows you to use StrictNewVersion() this means that we won't match against some of the non semver tags in on the lifecycle image (ex. 9832537 in the original branch parses as 9832537.0.0 using the StrictNewVersion() function errors out on this tag). 

I could switch over to using a Regex but this felt more straight-forward

Comment (ryanmoran) [reply go.mod:8]: I guess the question then is why do we need to use the older version elsewhere? I guess I'm just trying to see why we can't just use v3 everywhere.

Comment (ForestEckhardt) [reply go.mod:8]: That's fair. I would just have to go around and convert everything and see if I can just swap everything over.

Comment (ForestEckhardt) [reply go.mod]: @ryanmoran Alright, the latest commit should have upgraded everything to the v3 branch.

**Review by ryanmoran — APPROVED** [review]

**Pull request merged**